### PR TITLE
feat: add arguments --build-cpu and --build-memory when deploying apps

### PR DESCRIPTION
**Pull request by smile37773**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by smile37773** [comment]
Will update version in https://github.com/Azure/azure-cli-extensions/pull/4844

File: src/spring/azext_spring/tests/latest/test_asa_scenario.py [hunk 1]
```diff
@@ -218,12 +218,3 @@ def test_app_deploy_container(self):
...
             self.check('name', 'green'),
         ])
-
-        with self.assertRaisesRegexp(CLIError, "Failed to wait for deployment instances to be ready"):
-            self.cmd('spring app deploy -g {resourceGroup} -s {serviceName} -n {app} --artifact-path {file}')
-
-        self.cmd('spring app deployment show -g {resourceGroup} -s {serviceName} -n default --app {app} ', checks=[
-            self.check('name', 'default'),
-            self.check('properties.source.type', 'Jar'),
-            self.check('properties.source.runtimeVersion', 'Java_8'),
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/tests/latest/test_asa_scenario.py:228]: Why did you remove these tests?

**Comment by zhoxing-ms** [comment]
Could you add some tests for those new parameters?

File: src/spring/azext_spring/tests/latest/test_asa_scenario.py (hunk 1 above)
Comment (smile37773) [reply src/spring/azext_spring/tests/latest/test_asa_scenario.py:228]: Because it's meaningless for CustomImageTest. These commands are not used to deploy app with customer image but jar.

**Review by zhoxing-ms — APPROVED** [review]
